cdap-source-control/src/main/java/io/cdap/cdap/sourcecontrol/RepositoryManager.java

```diff
@@ -97,6 +103,46 @@ public static void validateConfig(SecureStore secureStore, SourceControlConfig s
     }
   }
 
+  /**
+   * Returns the <a href="https://git-scm.com/docs/git-hash-object">Git Hash</a>
+   * of the requested file path in the current HEAD. For symlinks, it returns the hash of
+   * the target file. This ensures that the file hash of a symlink is equal to the hash of the target file. It
+   * doesn't return the hash for directories. If the directory contains symlinks, we can't depend on the
+   * directory hash to ensure the contents are unchanged.
+   *
+   * @param path The absolute path on the filesystem that should include {@link RepositoryManager#getBasePath()} as a
+   *             prefix.
+   * @return The git file hash of the requested file.
+   * @throws IOException              when file isn't found, there are circular symlinks or other file IO errors.
+   * @throws IllegalArgumentException when the provided file isn't a regular file, eg: directory.
+   * @throws NotFoundException        when the file isn't committed to Git.
+   * @throws IllegalStateException    when {@link RepositoryManager#startUp()} isn't called before this.
+   */
+  public String getFileHash(Path path) throws IOException, NotFoundException {
```

Comment:
wondering if we could extend this call to accept a collection of paths and a commitID and return a list of fileHash?

Reply:
Sure can you make the changes method signature in the push operation PR? I'll implement the same method.

Reply:
BTW: Any specific reason we are using `Path`? Looking at the code this is absolute path and this exposes internal implementation (the fact that we actually have working copy in specified folder). Should we just use a path relative to the git repo and enforce it? This would also be helpful if we would need to have multiple repos at some point (multithreaded operations, other storage)?

Reply:
In other words, having an absolute path prevents from different workers having different base path

Reply:
@arjan-bal second thought while I added the batch getFileHash signature in [PR](https://github.com/cdapio/cdap/pull/14911), I'll keep it as a single hash signature for below reasons:
1. Because we're also retuning the `PushAppResponse` which has appName + version, it's not convenient to get that matched hash from the list result of getFileHash (might need to be a map?). 
2. There's no much computation in getFileHash since we'll pass in the `RevCommit` that has everything in it. Little benefits can be gained from the batch change
3. Also there's ongoing discussion of using relative path string from Vitalii, let's resolve that first

Reply:
@tivv Changed gitFIleHash to accept a relative path. Presently the file operations are done by the OperationRunner and repository manager does VCS stuff like commit, clone etc. So the RepositoryManage#getBasePath method is used by operation runner to know where to read/write files.